### PR TITLE
Migrations

### PR DESCRIPTION
**Pull request by FinnStutzenstein**
Basic setup for developing migrations, see the README.md in the
migrations folder

### CODE REVIEW

File: migrations/migrate.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        event: RawEvent,
+        old_accessor: MigrationKeyframeAccessor,
+        new_accessor: MigrationKeyframeAccessor,
+        position_data: PositionData,
+    ) -> List[RawEvent]:
+        return [event]
+
+def main():
+    handler = setup()
+    handler.register_migrations([FirstMigration])
```
Comment (jsangmeister) [inline migrations/migrate.py:19]: Decorator would be more comfortable here

**Comment by FinnStutzenstein** [comment]
TODO:
- Add new datastore exception: INVALID_DATASTORE_STATE

**Comment by FinnStutzenstein** [comment]
@jsangmeister Please review this, too. Whats missing are docs; I have to centralize the migration docs in the wiki (e.g. what is `migrate`, `finalize`, ...) and adapt the docs here and in the DS. Will do this later on.

**Comment by FinnStutzenstein** [comment]
TODO:
- CLI descriptions

**Review by jsangmeister — APPROVED** [review]
Looks really good!

**Pull request merged**